pkg/platform/kube/platform.go

```diff
@@ -681,6 +697,24 @@ func (p *Platform) GetFunctionReplicaNames(ctx context.Context,
 	return names, nil
 }
 
+func (p *Platform) GetFunctionAllReplicaNames(ctx context.Context, function platform.Function, permissionOptions opa.PermissionOptions, timeFilter *platform.TimeFilter) ([]string, error) {
+
+	functions, err := p.Platform.FilterFunctionsByPermissions(ctx, &permissionOptions, []platform.Function{function})
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to filter functions by permissions")
+	}
+
+	if len(functions) == 0 {
+		// Function was filtered out by permissions, return not found error
+		return nil, nuclio.NewErrNotFound(fmt.Sprintf("Function not found - %s", function.GetConfig().Meta.Name))
+	}
+
+	return p.elasticSearchClient.GetFunctionReplicas(ctx, &logProxy.GetFunctionReplicaOptions{
+		TimeFilter:   timeFilter,
+		FunctionName: function.GetConfig().Meta.Name,
+	})
```

Comment:
What if the elasticSearchClient doesn't exist?
it is only initialized `if platformConfiguration.Kube.ElasticSearchConfig != nil`. If it isn't initialized than you can return an empty array / nil at the beginning of this function.

Also - what if we have a different client (for a different source)?

Reply:
@TomerShor changed it to not fail when defaultProxyLogSource is not ES